client.go

```diff
@@ -91,167 +72,88 @@ func (c *Client) Dial(address string) (conn *Conn, err error) {
 		}
 	}
 
-	conn = new(Conn)
+	sync := make(chan bool)
+	clientConn = &ClientConn{srv: &Server{Net: network, TLSConfig: c.TLSConfig, ReadTimeout: c.readTimeout(), WriteTimeout: c.writeTimeout(), UDPSize: c.UDPSize,
+		NotifyStartedFunc: func() {
+			sync <- true
```

Comment:
i would add switch with timeout logging Fatal to yell that no one is listening maybe. 
```go
switch{
 case <-sync:
 case <- time.After(3 * time.Minute): log.Fatal("something")
}
```

sample might be incorrect, it's just from top of my head

Reply:
Fixed